File: base/src/main/java/com/example/base/presenter/IBasePresenter.java
package com.example.base.presenter;

import com.example.base.view.IBaseView;

/**
 * 获取数据, 各个页面都可以调用该接口获取数据
 */
public interface IBasePresenter<T> {
    //requestCode 来区分不同的网络请求
    void doHttpRequest(int requestCode);
    void doHttpPostRequest(int requestCode);
    void attachView(IBaseView<T> iBaseView); //关联IBaseView
    void detachView(); //解除关联

//    void getData(); //从net包中拿数据

}

File: base/src/main/java/com/bw/base/IBasePresenter.kt
package com.bw.base

//获取数据，各个页面都可以调用该接口获取数据
interface IBasePresenter<T> {

    //来区分不同的网络请求
    fun doHttpRequest()
    fun doHttpPostRequest()
    fun attachView(iBaseView:IBaseView<T>)
    fun datachView()

}